File: example-base-27/java-interview-utilities-complete/src/main/java/com/prem/anagram/BinarySearch.java
package com.prem.anagram;
public class BinarySearch {
    public static int search(int[] sortedNumbers,int target) {
        if(sortedNumbers==null) return -1;
        int left=0,right=sortedNumbers.length-1;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(sortedNumbers[mid]==target)return mid;
            if(sortedNumbers[mid]<target)left=mid+1;else right=mid-1;
        }
        return -1;
    }
}

File: example-base-28/nodejs-utilities-playground-v2/src/dsa/searching/binary-search.js
function binarySearch(numbers, target) {
  let left = 0, right = numbers.length - 1;
  while (left <= right) {
    const mid = left + Math.floor((right - left) / 2);
    if (numbers[mid] === target) return mid;
    if (numbers[mid] < target) left = mid + 1;
    else right = mid - 1;
  }
  return -1;
}
module.exports = { binarySearch };
